handsontable/src/core/focusCatcher/index.js

```diff
@@ -0,0 +1,136 @@
+      keys: [['Tab'], ['Shift', 'Tab']],
+      callback: (event) => {
+        const { disableTabNavigation, autoWrapRow } = hot.getSettings();
+
+        if (disableTabNavigation) {
+          hot.deselectCell();
+          hot.unlisten();
+
+          return false;
+        }
+
+        const isSelected = hot.selection.isSelected();
+        const highlight = hot.getSelectedRangeLast()?.highlight;
+        const mostTopStartCoords = getMostTopStartPosition(hot);
+        const mostBottomEndCoords = getMostBottomEndPosition(hot);
+
+        // For disabled `autoWrapRow` option set the row to the same position as the currently selected row.
+        if (!autoWrapRow) {
+          mostTopStartCoords.row = highlight.row;
+          mostBottomEndCoords.row = highlight.row;
+        }
+
+        if (event.shiftKey && (!isSelected || highlight.isEqual(mostTopStartCoords)) ||
+            !event.shiftKey && (!isSelected || highlight.isEqual(mostBottomEndCoords))) {
+          hot.deselectCell();
+          hot.unlisten();
+
+          return false;
+        }
+
+        return true;
+      },
+      runOnlyIf: () => !hot.getSettings().minSpareCols,
+      preventDefault: false,
+      stopPropagation: false,
+      position: 'before',
+      relativeToGroup: GRID_GROUP,
+      group: 'focusCatcher',
+    });
+}
+
+/**
+ * Gets the coordinates of the most top-start cell or header (depends on the table settings and its size).
+ *
+ * @param {Core} hot The Handsontable instance.
+ * @returns {CellCoords|null}
+ */
+function getMostTopStartPosition(hot) {
+  const { rowIndexMapper, columnIndexMapper } = hot;
+  const { navigableHeaders } = hot.getSettings();
+  let topRow = navigableHeaders && hot.countColHeaders() > 0 ? -hot.countColHeaders() : 0;
+  let startColumn = navigableHeaders && hot.countRowHeaders() > 0 ? -hot.countRowHeaders() : 0;
+
+  topRow = topRow === 0 ? rowIndexMapper.getVisualFromRenderableIndex(topRow) : topRow;
+  startColumn = startColumn === 0 ? columnIndexMapper.getVisualFromRenderableIndex(startColumn) : startColumn;
+
+  if (topRow === null || startColumn === null) {
+    return null;
+  }
+
+  return hot._createCellCoords(topRow, startColumn);
+}
+
+/**
+ * Gets the coordinates of the most bottom-end cell or header (depends on the table settings and its size).
+ *
+ * @param {Core} hot The Handsontable instance.
+ * @returns {CellCoords|null}
+ */
+function getMostBottomEndPosition(hot) {
+  const { rowIndexMapper, columnIndexMapper } = hot;
+  const { navigableHeaders } = hot.getSettings();
+  let bottomRow = rowIndexMapper.getRenderableIndexesLength() - 1;
+  let endColumn = columnIndexMapper.getRenderableIndexesLength() - 1;
+
+  if (bottomRow < 0) {
+    bottomRow = navigableHeaders && hot.countColHeaders() > 0 ? -1 : null;
+  }
+  if (endColumn < 0) {
+    endColumn = navigableHeaders && hot.countRowHeaders() > 0 ? -1 : null;
+  }
+
+  if (bottomRow === null || endColumn === null) {
+    return null;
+  }
```

Comment:
Up to you, for me it would be more clean.

```suggestion
  if (bottomRow < 0) {
    if (navigableHeaders === false || hot.countColHeaders() === 0) {
      return null;
    }

    bottomRow =-1;
  }

  if (endColumn < 0) {
    if (navigableHeaders === false || hot.countColHeaders() === 0) {
      return null;
    }

    endColumn = -1;
  }
```